core/storage/repository/impl/filesystem_repository.hpp

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef FILECOIN_CORE_STORAGE_IMPL_FILESYSTEM_REPOSITORY_HPP
+#define FILECOIN_CORE_STORAGE_IMPL_FILESYSTEM_REPOSITORY_HPP
+
+#include <iostream>
+
+#include "fslock/fslock.hpp"
+#include "storage/filestore/path.hpp"
+#include "storage/leveldb/leveldb.hpp"
+#include "storage/repository/repository.hpp"
+
+namespace fc::storage::repository {
+
+  using filestore::Path;
+  using Version = Repository::Version;
+
+  /**
+   * @brief FileSystem implementation of Repository.
+   *
+   *  Directory structure of Filesystem Repository is:
+   * .ipfs/
+   * ├── api             <--- running daemon api addr
+   * ├── blocks/         <--- objects stored directly on disk
+   * │   └── aa          <--- prefix namespacing like git
+   * │       └── aa      <--- N tiers
+   * ├── config          <--- config file (json or toml)
+   * ├── hooks/          <--- hook scripts (not implemented yet)
+   * ├── keys/           <--- cryptographic keys
+   * │   ├── id.pri      <--- identity private key
+   * │   └── id.pub      <--- identity public key
+   * ├── datastore/      <--- datastore
+   * ├── logs/           <--- 1 or more files (log rotate)
+   * │   └── events.log  <--- can be tailed
+   * ├── repo.lock       <--- mutex for repo
+   * └── version         <--- version file
+   *
+   */
+  class FileSystemRepository : public Repository {
+   public:
+    inline static const std::string kApiFilename = "api";
```

Comment:
do we really need `inline` here?

Reply:
Yes, see [this](https://www.codingame.com/playgrounds/2205/7-features-of-c17-that-will-simplify-your-code/inline-variables)